File: examples/hooks/python/user_prompt_prefix.py
#!/usr/bin/env python3
"""UserPromptSubmit — prefix every user prompt (Python)."""

from __future__ import annotations

import os
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parent))
from lib.hook_io import allow, modify_prompt, read_event  # noqa: E402

PREFIX = os.environ.get(
    "SOLCODE_PROMPT_PREFIX",
    "[project note: prefer small diffs; run tests after edits]\n\n",
)


def main() -> None:
    event = read_event()
    prompt = str(event.get("prompt") or "")
    if not prompt or prompt.startswith(PREFIX.strip()):
        allow()
        return
    modify_prompt(PREFIX + prompt, "prefixed user prompt")


if __name__ == "__main__":
    main()

File: examples/hooks/user-prompt-prefix.mjs
#!/usr/bin/env node
/**
 * UserPromptSubmit — prefix every user prompt with a project reminder
 *
 * {
 *   "hooks": {
 *     "events": {
 *       "UserPromptSubmit": [{
 *         "matcher": "*",
 *         "hooks": [{
 *           "type": "command",
 *           "command": "node examples/hooks/user-prompt-prefix.mjs",
 *           "timeout_ms": 3000,
 *           "fail_mode": "open"
 *         }]
 *       }]
 *     }
 *   }
 * }
 *
 * Note: matcher for UserPromptSubmit is matched against the prompt string
 * (exact segment match via "|"), so prefer matcher "*" here.
 */

import { readEvent, allow, modifyPrompt } from "./lib/read-event.mjs";

const PREFIX =
  process.env.SOLCODE_PROMPT_PREFIX ||
  "[project note: prefer small diffs; run tests after edits]\n\n";

const event = await readEvent();
const prompt = String(event.prompt || "");

if (!prompt || prompt.startsWith(PREFIX.trim())) {
  allow();
  process.exit(0);
}

modifyPrompt(PREFIX + prompt, "prefixed user prompt");
